mitiq/parameter.py

```diff
@@ -0,0 +1,257 @@
+from typing import Iterable
+import numpy as np
+
+import copy
+
+from cirq import Circuit, Gate, value, unitary, Moment, X, Z, Y
+from cirq import (
+    ZPowGate, YPowGate, XPowGate,
+    HPowGate, CXPowGate, CZPowGate,
+    MeasurementGate
+)
+from cirq.ops import gate_features
+
+from mitiq._typing import QPROGRAM
+from mitiq.folding import converter
+
+BASE_GATES = [ZPowGate, HPowGate, XPowGate, YPowGate, CXPowGate \
+            CZPowGate]
+
+@converter
+def scale_parameters(
+    circ: Circuit,
+    scale_factor: float,
+    sigma: float,
+    seed: int = None
+) -> Circuit:
+    """Adds parameter noise to a circuit with level noise.
+    This adds noise to the actual parameter instead of
+    adding an parameter channel.
+
+    Args:
+        circ: The quantum program as a Cirq circuit object. All measurements
+            should be in the last moment of the circuit.
+        scale_factor: Amount to scale the base noise level of parameter rotations by.
+        sigma: Base noise level (variance) in parameter rotations
+        seed: 
+
+    Returns:
+        The input circuit with scaled rotation angles
+
+    """
+    final_moments = []
+    noise = (scale_factor - 1) * sigma
+    for moment in circ:
+        curr_moment = []
+        for op in moment.operations:
+            gate = copy.deepcopy(op.gate)
+            qubits = op.qubits
+            if isinstance(gate, MeasurementGate):
+                curr_moment.append(gate(*qubits))
+            else:
+                base_gate = _get_base_gate(gate)
+                param = gate.exponent * np.pi
+                rng = np.random.RandomState(seed)
```

Comment:
Should be defined before the loop otherwise `rng.normal` will return the same value every time.